resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -162,6 +161,8 @@ private[spark] class Client(
                   .endEnv()
                 .addToEnv(sslEnvs: _*)
                 .withPorts(containerPorts.asJava)
+                .withNewReadinessProbe().withHttpGet(probePingHttpGet).endReadinessProbe()
+                .withNewLivenessProbe().withHttpGet(probePingHttpGet).endLivenessProbe()
```

Comment:
do we need `initialDelaySeconds` / `timeoutSeconds` / `periodSeconds` set here?

https://kubernetes.io/docs/api-reference/v1/definitions/#_v1_probe

I'm also not sure that having both is necessary -- we only rely on the readiness probe in this PR?